register/views.py

```diff
@@ -1,8 +1,68 @@
-from registration.backends.default.views import ActivationView
+from django.views.generic import TemplateView, FormView, RedirectView
+from django.shortcuts import render
+from django.contrib.auth import get_user_model, login, logout
+from django.urls import reverse, reverse_lazy
+
 from registration.backends.default.views import RegistrationView
 
-from django.views.generic import TemplateView
+from . import forms
 
 
 class ProfileView(TemplateView):
     template_name = 'register/profile.html'
+
+
+class EmailView(FormView):
+    template_name = 'registration/email.html'
+    success_url_login = reverse_lazy('auth_login')
+    success_url_register = reverse_lazy('auth_register')
+    form_class = forms.EmailForm
+
+    def form_valid(self, form):
+        self.request.session['email'] = form.data.get('email')
```

Comment:
We must provide form.data['email'] otherwise, when POSTed, the form processes the email as empty and is considered invalid.